tidy3d/plugins/smatrix/ports/rectangular_lumped.py

```diff
@@ -0,0 +1,265 @@
+        center = list(self.center)
+        if snap_center:
+            center[self.injection_axis] = snap_center
+
+        e_component = "xyz"[self.voltage_axis]
+        # Size of voltage monitor can essentially be 1D from ground to signal conductor
+        voltage_mon_size = list(self.size)
+        voltage_mon_size[self.injection_axis] = 0.0
+        voltage_mon_size[self.current_axis] = 0.0
+        # Create a voltage monitor
+        return FieldMonitor(
+            center=center,
+            size=voltage_mon_size,
+            freqs=freqs,
+            fields=[f"E{e_component}"],
+            name=f"{self.name}_voltage",
+            colocate=False,
+        )
+
+    def to_current_monitor(self, freqs: FreqArray, snap_center: float) -> FieldMonitor:
+        """Field monitor to compute port current."""
+        center = list(self.center)
+        if snap_center:
+            center[self.injection_axis] = snap_center
+
+        h_component = "xyz"[self.current_axis]
+        h_cap_component = "xyz"[self.injection_axis]
+        # Size of current monitor needs to encompass the current carrying 2D sheet
+        # Needs to have a nonzero thickness so a closed loop of gridpoints around the 2D sheet can be formed
+        dl = 2 * (increment_float(center[self.injection_axis], 1.0) - center[self.injection_axis])
+        current_mon_size = list(self.size)
+        current_mon_size[self.injection_axis] = dl
+        current_mon_size[self.voltage_axis] = 0.0
+        # Create a current monitor
+        return FieldMonitor(
+            center=center,
+            size=current_mon_size,
+            freqs=freqs,
+            fields=[f"H{h_component}", f"H{h_cap_component}"],
+            name=f"{self.name}_current",
+            colocate=False,
+        )
+
+    def compute_voltage(self, sim_data: SimulationData) -> FreqDataArray:
+        """Helper to compute voltage across the port."""
+        field_data = sim_data[f"{self.name}_voltage"]
+        size = list(self.size)
+        size[self.current_axis] = 0
+        voltage_integral = VoltageIntegralAxisAligned(
+            center=self.center,
+            size=size,
+            extrapolate_to_endpoints=True,
+            snap_path_to_grid=True,
+            sign="+",
+        )
+        voltage = voltage_integral.compute_voltage(field_data)
+        # Return data array of voltage with coordinates of frequency
+        return voltage
+
+    def compute_current(self, sim_data: SimulationData) -> FreqDataArray:
+        """Helper to compute current flowing through the port."""
+        # Diagram of contour integral, dashed line indicates location of sheet resistance
+        # and electric field used for voltage computation. Voltage axis is out-of-page.
+        #
+        #                                    current_axis = ->
+        #                                    injection_axis = ^
+        #
+        #                  |                   h2_field ->             |
```

Comment:
should we include this in the docstring?

Reply:
Is there a way to keep the white space formatting? It doesn't really look nice

Reply:
hm, yea not sure. I guess we can just leave it as a code comment.